turtle.go

```diff
@@ -14,21 +15,22 @@ const (
 	ErrNotWriteTxn = errors.Error("cannot perform write actions during a read transaction")
 	// ErrKeyDoesNotExist is returned when a key does not exist
 	ErrKeyDoesNotExist = errors.Error("key does not exist")
+	// ErrEmptyKey is returned when an empty key is provided
+	ErrEmptyKey = errors.Error("empty keys are invalid")
 )
 
 // Value is the value type
 type Value generic.Type
 
 // New will return a new instance of Turtle
-func New(name, path string, mfn MarshalFn, ufn UnmarshalFn) (tp *Turtle, err error) {
+func New(name, path string, fm FuncsMap) (tp *Turtle, err error) {
 	var t Turtle
-	if t.mrT, err = mrT.New(path, name); err != nil {
+	if t.mrT, err = mrT.New(path, name, middleware.Base64MW{}); err != nil {
 		return
 	}
 
-	t.s = make(store)
-	t.mfn = mfn
-	t.ufn = ufn
+	t.b = newBuckets()
+	t.fm = fm
```

Comment:
IMHO it should have a sane default of fm is nil, for example use json by default.

```
if fm == nil {
fm = FuncsMap{default:json.*}
}
```